csvcubed/csvcubed/readers/columnschema.py

```diff
@@ -0,0 +1,447 @@
+"""
+Models
+------
+
+config.json V1.0 column mapping models.
+"""
+from dataclasses import dataclass
+from typing import List, Union, Optional, TypeVar
+from pathlib import Path
+
+import uritemplate
+from csvcubed.models.cube import (
+    ExistingQbAttributeLiteral,
+    NewQbAttributeLiteral,
+    CompositeQbCodeList,
+    CatalogMetadata,
+    DuplicatedQbConcept,
+    NewQbCodeList,
+)
+from csvcubedmodels.dataclassbase import DataClassBase
+
+from csvcubed.inputs import pandas_input_to_columnar_optional_str
+from csvcubed.models.cube.qb.components import (
+    NewQbDimension,
+    ExistingQbDimension,
+    NewQbAttribute,
+    NewQbAttributeValue,
+    ExistingQbAttribute,
+    NewQbUnit,
+    ExistingQbUnit,
+    QbMultiUnits,
+    QbMultiMeasureDimension,
+    QbMultiMeasureObservationValue,
+    QbSingleMeasureObservationValue,
+    ExistingQbCodeList,
+    ExistingQbMeasure,
+    NewQbMeasure,
+    QbObservationValue,
+    NewQbCodeListInCsvW,
+)
+from csvcubed.inputs import PandasDataTypes
+from csvcubed.utils.uri import looks_like_uri, csvw_column_name_safe
+
+T = TypeVar("T", bound=object)
+
+
+@dataclass
+class SchemaBaseClass(DataClassBase):
+    ...
+
+
+@dataclass
+class Dimension(SchemaBaseClass):
```

Comment:
Even if we have one underlying model in the JSON schema, can we not split this into two models in python? One for Existing and another for New?

Reply:
This is contrary to previous guidance where you indicated that the schema classes should correspond to the json schema.
This is because there is no clear distinction between the object properties for NewDimension and ExistingDimension (etc.) 

For infojson the structure was clearly different with the 'new' property being present for New columns.
If this is really required then we need to discuss how to differentiate these.

Currently a call that is conditional on 'type' == 'dimension' to schema.**Dimension**.dict_fields_match_class confirms we have the correct fields before instantiating the Dimension dataclass, the branching to choose New/Existing is done in the map_to_qb_dimension method.

Feasibly we could set type to new_dimension or exsiting_dimension on the config json so the conditional logic containing the dict_fields_match_class could be updated to check the correct New/Existing dimension fields.

Alternatively the schema itself could be updated so that the properties contain either/and title or description to be considered a new dimension.  For other column types this needs checking.